app/deploy.go

```diff
@@ -252,6 +252,9 @@ func deployToProvisioner(opts *DeployOptions, evt *event.Event) (string, error)
 	if opts.Kind == "" {
 		opts.GetKind()
 	}
+	if ((opts.App.GetPlatform() == "") && (opts.Kind != DeployImage)) || ((opts.App.GetPlatform() == "") && (opts.Kind != DeployRollback)) {
```

Comment:
I think you can write this as:
`((opts.App.GetPlatform() == "") && (opts.Kind != DeployImage && opts.Kind != DeployRollback))`